activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/conf/EventHandlersAutoConfiguration.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.cloud.conf;
+
+import org.activiti.cloud.services.query.app.repository.ProcessDefinitionRepository;
+import org.activiti.cloud.services.query.app.repository.ProcessModelRepository;
+import org.activiti.cloud.services.query.events.handlers.ProcessDeployedEventHandler;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class EventHandlersAutoConfiguration {
+
+    @Bean
```

Comment:
Is there a reason for not including any of the other handlers in the autoconfiguration https://github.com/Activiti/activiti-cloud-query-service/tree/develop/activiti-cloud-services-query/activiti-cloud-services-query-rest/src/main/java/org/activiti/cloud/services/query/events/handlers ?

I agree [autoconfiguration is the way forward](https://github.com/Activiti/Activiti/issues/2258) but find it misleading to have a class called EventHandlersAutoConfiguration when it is only loading one handler and everything else is coming from a componentscan (which [actually is more general and would take priority anyway](https://github.com/Activiti/activiti-cloud-query-service/blob/b6ff8c630fb4d171188b4262bd4bccffd8ee10e4/activiti-cloud-starter-query/src/main/java/org/activiti/cloud/starter/query/configuration/ActivitiQueryAutoConfiguration.java#L8)).

Reply:
Totally agree @ryandawsonuk. The idea of this auto-configuration is to include all the other handlers as well, but currently this was the only one using auto-configuratoon approach: I only extracted it from `QueryRestAutoConfiguration` to a new class for commodity in controllers tests. But you're right, maybe it's worth to spent some extra time converting other handlers to auto-configuration approach so we don't need to come back to it later on. I'll do that.